### PR TITLE
Update the metadata API

### PR DESCRIPTION
**Pull request by rebornix**
Update the metadata API with upstream

### CODE REVIEW

File: src/client/common/application/notebook.ts [hunk 1]
```diff
@@ -94,12 +94,19 @@ export class VSCodeNotebook implements IVSCodeNotebook {
...
-        return notebook.registerNotebookKernelProvider(selector, provider);
+
+    // IANHU
+    // public registerNotebookKernelProvider(
+    // selector: NotebookDocumentFilter,
+    // provider: NotebookKernelProvider
+    // ): Disposable {
+    // return notebook.registerNotebookKernelProvider(selector, provider);
+    // }
+    //createNotebookKernel(options: NotebookKernelOptions): NotebookKernel2;
```
Comment (DonJayamanne) [inline src/client/common/application/notebook.ts:105]: Can we not leave the old code in tact?
Or is the push model implementation of kernels complete?

Comment (IanMatthewHuff) [reply src/client/common/application/notebook.ts:105]: No, this was an old half done branch. I think it was just based on that to get the proposed API right.

File: src/client/datascience/kernel-launcher/localKernelFinder.ts [hunk 2]
```diff
@@ -176,13 +180,19 @@ export class LocalKernelFinder implements ILocalKernelFinder {
...
     ): Promise<LocalKernelConnectionMetadata[]> {
+        traceInfo('IANHU');
         // First find the on disk kernel specs and interpreters
-        const [kernelSpecs, interpreters, rootSpecPath, activeInterpreter] = await Promise.all([
-            this.findResourceKernelSpecs(resource, cancelToken),
-            this.findResourceInterpreters(resource, cancelToken),
-            this.getKernelSpecRootPath(),
-            this.getActiveInterpreter(resource)
-        ]);
+        const activeInterpreter = await this.getActiveInterpreter(resource);
```
Comment (rchiodo) [inline src/client/datascience/kernel-launcher/localKernelFinder.ts:185]: Uh not sure why you'd do these in order instead of at the same time?

Comment (rchiodo) [reply src/client/datascience/kernel-launcher/localKernelFinder.ts:185]: Was something not working?

Comment (IanMatthewHuff) [reply src/client/datascience/kernel-launcher/localKernelFinder.ts:185]: @rchiodo Sorry you are not on the teams chat. It's based off a junk branch that I was doing some kernel work in, just to get the correct proposed API changes (which I had already done in that branch). I'm going to port this after my breaking changes fix is in.

**Review by IanMatthewHuff — COMMENTED** [review]
@rebornix I don't think there is anything in here that I don't have in this PR:
https://github.com/microsoft/vscode-jupyter/pull/5530/files
See anything that I'm missing?

**Comment by IanMatthewHuff** [comment]
Closing as this is covered in the other PRs. Thanks for the help @rebornix

**Pull request closed without merging**